R/util-cli_logger_wrappers.R

```diff
@@ -0,0 +1,55 @@
+#' cli style console appender for gsm
+#'
+#' @param level warning level that maps to log4r
+#' @param ... should contain message and cli_detail
+#'
+#' @export
+cli_fmt <- function(level, ...) {
+  fields <- list(...)
+  if (level == "INFO" && fields$cli_detail == "h1") {
+    cli::cli_h1(fields$message)
+  } else if (level == "INFO" && fields$cli_detail == "h2") {
+    cli::cli_h2(fields$message)
+  } else if (level == "INFO" && fields$cli_detail == "h3") {
+    cli::cli_h3(fields$message)
+  } else if (level == "INFO" && fields$cli_detail == "alert") {
+    cli::cli_alert(fields$message)
+  } else if (level == "INFO" && fields$cli_detail == "alert_success") {
+    cli::cli_alert_success(fields$message)
+  } else if (level == "INFO" && fields$cli_detail == "alert_info") {
+    cli::cli_alert_info(fields$message)
+  } else if (level == "WARN") {
+    cli::cli_alert_warning(fields$message)
+  } else if (level == "ERROR") {
+    cli::cli_alert_danger(fields$message)
+  } else {
+    cli::cli_inform(fields$message)
+  }
+  return(NULL)
+}
+
+#' Custom logging function that wraps cli messaging
+#' @param level logger levels
+#' @param message message to display; may contain glue-style placeholders
+#' @param cli_detail for cli style alerts the detail for info
+#' @param .envir the environment for glue expressions
+#'
+#' @export
+LogMessage <- function(level, message, cli_detail = NULL, .envir = parent.frame()) {
```

Comment:
@lauramaxwell ^ before I proliferate the use of this function for all our `cli` calls, we essentially now have a `logger` object as part of our package with a "custom layout" that essentially wraps/calls `cli`, but if a user/gismo wants a different logging methodology, it can use that instead by swapping in a different `logger` object using `SetLogger()`

Reply:
ah, okay. i think this makes sense- so this just formats the message in the same way we currently display, but `LogMessage()` can be swapped out easily if we want.

Reply:
To "see" it,

Load this version of the package, try running Step 1, the mapping workflow from example 3.1.

Then run:
```
SetLogger(logger = logger(appender = console_appender(layout = json_log_layout())))
```

and run step 1 example 3.1 again